src/net_processing.cpp

```diff
@@ -3114,8 +3118,28 @@ bool static ProcessMessage(CNode* pfrom, const std::string& strCommand, CDataStr
     }
 
     if (strCommand == NetMsgType::NOTFOUND) {
-        // We do not care about the NOTFOUND message, but logging an Unknown Command
-        // message would be undesirable as we transmit it ourselves.
+        // NOTFOUND means we can clear the entry from the tx_in_flight/announced set
+        std::vector<CInv> vInv;
+        vRecv >> vInv;
+        // note: we don't enforce a limit on number of inv's in a notfound
+
+        LogPrint(BCLog::NET, "received notfound (%u invsz) peer=%d\n", vInv.size(), pfrom->GetId());
+
+        if (vInv.size() > 0) {
+            LogPrint(BCLog::NET, "received notfound for: %s peer=%d\n", vInv[0].ToString(), pfrom->GetId());
```

Comment:
Could you explain what's going on here? Why we're logging only the first notfound INV?

Reply:
It's doing the same logging as `getdata`. which also only logs the first entry. Didn't seem terribly valuable to log every notfound entry seen.